cnf-tests/testsuites/e2esuite/metrics/sriovnetworkmetricsexporter.go

```diff
@@ -0,0 +1,268 @@
+package metrics
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	sriovtestclient "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/client"
+	sriovcluster "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/cluster"
+	sriovnamespaces "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/namespaces"
+	sriovnetwork "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/network"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/images"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/networks"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/pods"
+
+	"github.com/prometheus/common/model"
+)
+
+const testNamespace string = "test-sriov-metrics"
+
+var sriovclient *sriovtestclient.ClientSet
+
+func init() {
+	sriovclient = sriovtestclient.New("")
+}
+
+var _ = Describe("[sriov] SR-IOV Network Metrics Exporter", func() {
+
+	var sriovCapableNodes *sriovcluster.EnabledNodes
+
+	BeforeEach(func() {
```

Comment:
should this one be `BeforeAll` if in the future we add more tests?

Reply:
Sounds good!